### ISSUE TITLE
The build script of hdf5-sys reports error with LLVM lld

### ISSUE THREAD

**Issue by Berrysoft** [metno/hdf5-rust#16]
https://github.com/metno/hdf5-rust/blob/41d0221cc2340b6c94f7ac5fde17a9ec89576818/hdf5-sys/build.rs#L127

It assumes that the command `ld --verbose` would success, but LLVM lld would just report error:
```
$ ld --verbose
ld: error: no input files
```

**Comment by magnusuMET** [comment]
Which platform is this on?

**Comment by Berrysoft** [comment]
Gentoo Linux. I have found the real problem: the hdf5 package on Gentoo doesn't contain a `libhdf5.pc` for pkg-config. I think it's the problem of Gentoo.

**Comment by magnusuMET** [comment]
We could have a fallback when pkg-config is not available, but for this I need to set up a platform etc.

**Comment by Berrysoft** [comment]
Yes, there's a fallback to get the library dirs from `ld --verbose`. But when the linker is `ld.lld`, the command fails.

**Comment by thesamesam** [comment]
In Gentoo, we still build hdf5 w/ autotools because of bugs in their CMake setup, but they don't install the pkgconfig file w/ autotools apparently: https://github.com/HDFGroup/hdf5/pull/2219.
